Makefile

```diff
@@ -14,16 +14,27 @@ endif
 help: ## List available commands
 	@cat $(MAKEFILE_LIST) | grep -E '^[a-zA-Z_-]+:.*?## .*$$' | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-.PHONY: requirements
-requirements: venv ## Install requirements
-	${VIRTUALENV_ROOT}/bin/pip install -Ur requirements.txt
-
 .PHONY: venv
 venv: ${VIRTUALENV_ROOT}/activate ## Create virtualenv if it does not exist
 
 ${VIRTUALENV_ROOT}/activate:
 	[ -z $$VIRTUAL_ENV ] && [ ! -d venv ] && python3 -m venv venv || true
 
+.PHONY: requirements
+requirements: venv ## Install requirements
+	${VIRTUALENV_ROOT}/bin/pip install -Ur requirements.txt
+
+.PHONY: requirements-test
+requirements-test: requirements ## Install test requirements
+	${VIRTUALENV_ROOT}/bin/pip install -Ur requirements-jenkins-job-builder.txt
+ifeq (, $(shell which jq))
+	$(error jjb-lint.sh requires jq, please ensure it is installed)
+endif
+
+.PHONY: test
+test: requirements-test
+	source ${VIRTUALENV_ROOT}/bin/activate && tools/jjb-lint/jjb-lint.sh job_definitions/*.yml
```

Comment:
Dunno if you can use the existing venv command on l17 here?

Reply:
I think l17 installs the virtualenv but doesn't activate it.